### PR TITLE
JAMES-4083 remove logback jackson usages from james

### PR DESCRIPTION
**Pull request by florentos17**
cf the [jira ticket](https://issues.apache.org/jira/browse/JAMES-4083)

i mounted the jar files as suggested, but i could not manage to point to them in every pom file. using ${project.basedir} in the parent pom does not work since it points to each pom's individual base directory... should i specify the path in each pom ?

### CODE REVIEW

**Comment by chibenwa** [comment]
```
[WARNING] 'dependencyManagement.dependencies.dependency.systemPath' for ch.qos.logback.contrib:logback-jackson:jar should not point at files within the project directory, ${project.basedir}/root/libs/logback-jackson-0.1.5.jar will be unresolvable by dependent projects @ line 2082, column 29
[WARNING] 'dependencyManagement.dependencies.dependency.systemPath' for ch.qos.logback.contrib:logback-json-classic:jar should not point at files within the project directory, ${project.basedir}/root/libs/logback-jackson-classic-0.1.5.jar will be unresolvable by dependent projects @ line 2089, column 29
[WARNING] 'dependencyManagement.dependencies.dependency.systemPath' for ch.qos.logback.contrib:logback-json-classic:jar refers to a non-existing file /home/jenkins/workspace/james_ApacheJames_PR-2482/root/libs/logback-jackson-classic-0.1.5.jar @ line 2089, column 29
```

@florentos17  this PR is opened for over 2 weeks and failures are clearly related.

Please take some time on it!

File: pom.xml [hunk 1]
```diff
@@ -2078,11 +2078,15 @@
                 <groupId>ch.qos.logback.contrib</groupId>
                 <artifactId>logback-jackson</artifactId>
                 <version>0.1.5</version>
+                <scope>system</scope>
+                <systemPath>${project.basedir}/root/libs/logback-jackson-0.1.5.jar</systemPath>
```
Comment (chibenwa) [inline pom.xml:2082]: Remove to not mean add the linagora specific lib as a magic jar depedency.

BTW binary is not reviewable source code and do not meet ASF standards.

**Review by chibenwa — CHANGES_REQUESTED** [review]

**Comment by florentos17** [comment]
so the tests failing are all of those from the class `james-project/third-party/crowdsec/src/test/java/org/apache/james/crowdsec/CrowdsecIntegrationTest.java`

i run those locally and they passed without a problem...!

**Comment by chibenwa** [comment]
rebase?

**Review by Arsnael — APPROVED** [review]

**Comment by chibenwa** [comment]
And update the lockat files to remove mentions of Json encoders.

**Comment by chibenwa** [comment]
This indeed is more fundamental.

Crowdsec extension as it exists relies on structured logging.

CF https://github.com/apache/james-project/blob/fa5f1c8484901fad629a13cb209c515717e7489a/third-party/crowdsec/sample-configuration/parsers/syslog-logs.yaml#L40

My proposal would be to:

 - Add logback-json back as a provided dependency of `third-party/crowdsec`
 - And document how to mount in runtime the `logback-json` file onto the classpath.

File: third-party/crowdsec/README.md [hunk 2]
```diff
@@ -17,6 +17,12 @@ This module is for developing and delivering extensions to James for the [Crowds
 guice.extension.module=org.apache.james.crowdsec.module.CrowdsecModule
 ```
 
+- Mount the configuration file onto the classpath
```
Comment (chibenwa) [inline third-party/crowdsec/README.md:20]: How to do this?

**Review by chibenwa — APPROVED** [review]

**Comment by chibenwa** [comment]
(already done) 

https://github.com/linagora/tmail-backend/pull/1333

**Pull request merged**